python/pyspark/pandas/frame.py

```diff
@@ -5495,7 +5506,15 @@ def op(psser: ps.Series) -> ps.Series:
 
         inplace = validate_bool_kwarg(inplace, "inplace")
         if inplace:
-            self._update_internal_frame(psdf._internal, requires_same_anchor=False)
+            # Since Pandas 1.4, df.fillna generates a new dataframe instead of operating
+            # in-place. See also SPARK-38946. Currently, `bfill` and `ffill` still keep
+            # inplace operate behaviors.
+            force_disconnect = (method is None) and LooseVersion(pd.__version__) >= LooseVersion(
```

Comment:
Spark 3.4 is claimed to match pandas 1.4, do we need this version check?

Reply:
It seems fine to keep it. I am just wondering about our principle for similar cases.

Reply:
Yep, this is a good question. Now it's more for make CI passed with current pandas version 1.3.x.

I think we should remove these code after we upgrade to 1.4.x. WDYT?

Reply:
Is keeping version checks in unit tests, but no version checks in main codes a good idea?

Reply:
Ohh sorry didn't refresh the page to see your comment. Sounds good!

Reply:
I think `keeping version checks in unit tests` is better, and also a better principle for similar cases:
- Don't use version check in code/implementation as far as possible, UT is a better place to do it, according UT we can easy to see what the behavior change.
- All test should be passed with any version of pandas > 1.0.5.